benchto-driver/src/main/java/io/trino/benchto/driver/listeners/profiler/perf/PerfProfiler.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.benchto.driver.listeners.profiler.perf;
+
+import io.trino.benchto.driver.listeners.profiler.QueryProfiler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+
+import static java.util.Objects.requireNonNull;
+
+@Component
+@ConditionalOnProperty(prefix = "benchmark.feature.profiler.perf", value = "enabled", havingValue = "true")
+public class PerfProfiler
```

Comment:
Could you document how it is working? what is required on the node/pod side?

Reply:
added entry to `README.me`